### DISCUSSION TITLE
Linux self compiled build missing tons of features

### DISCUSSION THREAD

**Discussion by YosypC** [opentrack/opentrack#1825] [Q&A]
I haven't found a previous discussion.
I've built my own OpenTrack following the Wiki, but it doesn't tell anywhere on how to include different Trackers.

![image](https://github.com/opentrack/opentrack/assets/50027752/de4d0b1b-386d-4434-beba-41b18f3b9858)
![image](https://github.com/opentrack/opentrack/assets/50027752/4026b98e-0f86-44b7-8ff6-60ca90e216c3)

For example, I need neuralnet as input, and the FreeTrack / TrackIR mock as output.

Please note I don't know much about Make.

EDIT: I got tons of GLOB mismatches after I "make install"

```
[...]
-- GLOB mismatch!
-- GLOB mismatch!
-- GLOB mismatch!
-- User settings file '/home/yosyp/Documents/Git/opentrack/opentrack/sdk-paths-yosyp@GNU-Linux.cmake' doesn't exist
-- module api
-- module compat
-- module csv
-- module cv
-- module filter-accela
-- module filter-ewma
-- module filter-hamilton
-- module filter-kalman
-- module filter-nm
-- module user-interface
-- module logic
-- module migration
-- module executable
-- module options
-- module pose-widget
-- module proto-fgfs
-- module proto-libevdev
-- module proto-udp
-- module qxt-mini
-- module spline
-- module tracker-easy
-- module tracker-freepie-udp
-- module tracker-fusion
-- module tracker-linux-joystick
-- module tracker-pt-base
-- module tracker-pt
-- module tracker-s2bot
-- module tracker-test
-- module tracker-udp
-- module video-opencv
-- module video
```

EDIT2: #1701 & #1539.
I'm on Mint, and there's no `onnxruntime-bin` .

**Comment by YosypC** [comment]
I solved by downloading [their](https://github.com/microsoft/onnxruntime/releases/tag/v1.17.1) runtime manually and ccmake the dir. But I still miss a way to output to my games.

How does it work, exactly, on Linux and Wine (Proton)?

**Comment by Coecoenut** [comment]
I run it with Star citizen. Choosing the "Wine -- Windows layer for Unix" as output in opentrack worked for me. There should be nothing else to do, but Proton/Steam or the game might require some additional settings.
Edit: After seeing that you don't have Wine as output option, you want to check this Reddit thread. It helped me to compile it with neuralnet and wine support: https://www.reddit.com/r/linux_gaming/comments/nlcg9n/comment/jn03qbg/

**Reply by YosypC** [reply]
I can't compile it with Wine.

#1826

**Comment by cgiAlexis** [comment]
So following what was said [here](https://github.com/opentrack/opentrack/issues/1539#issuecomment-1260479057), [here](https://github.com/opentrack/opentrack/issues/1701) and [here](https://github.com/opentrack/opentrack/discussions/1825#discussioncomment-8933538) I ended up downloading the premade files [from microsoft's repo](https://github.com/microsoft/onnxruntime/releases) as stated and extracting that somewhere.

### To generate the neuralnet tracker plugin;
1. Run `ccmake .` in the project root directory
2. Press `t` to enter advanced mode
3. Scroll down to `ONNXRuntime_DIR`, press enter, and paste the directory for the MS tracking software in
4. Pressing enter we scroll down and confirm `CMAKE_CXX_FLAGS` is set to `-fopenmp` apparently
5. Then `c` to configure, `e` to accept, `g` to generate
6. If we update the directory we need to change the directory three times under `ONNXRuntime_...` in Advanced settings then run again

### TrackIR output
1. `sudo apt install wine-development libwine-dev`
2. `ccmake .`
3. Enable WINE_SDK to ON
4. `c` to configure, `e` to accept, `g` to generate

I wonder if I set up some Actions for this project, if we couldn't have it spit out both a Linux package and have it generate the WINE .DLLs required to enable TrackIR in any program

**Reply by sthalik** [reply]
You can use this from the repository's root:
```
$ git grep 'set(SDK_' {tracker,proto,filter-video}-* CMakeLists.txt cmake/*.cmake
```

**Reply by cgiAlexis** [reply]
### Wine-development
Following what I saw here https://github.com/opentrack/opentrack/issues/1826#issuecomment-2028093544 I then used 
```bash
sudo apt install wine-development libwine-dev
```
 on Pop!_OS 22.04. I think I need to edit the wiki for you to add these dependencies. I didn't need to compile WINE locally at that point.

Sorry boss but I don't understand, running your code gives me the following but is that just a "for my information" on what trackers I should grab bits for? 
```bash
proto-fsuipc/CMakeLists.txt:    set(SDK_FSUIPC "" CACHE PATH "FSUIPC for older MS FSX path")
proto-osc/CMakeLists.txt:set(SDK_OSCPACK "" CACHE PATH "oscpack build directory")
proto-vjoystick/CMakeLists.txt:    set(SDK_VJOYSTICK "" CACHE PATH "vjoystick SDK path")
proto-wine/CMakeLists.txt:    set(SDK_WINE "" CACHE BOOL "Build for Wine")
tracker-aruco/CMakeLists.txt:    set(SDK_ARUCO_LIBPATH "" CACHE FILEPATH "Aruco paper marker tracker static library path")
tracker-eyeware-beam/CMakeLists.txt:set(SDK_EYEWARE_BEAM "" CACHE PATH "Eyeware Beam SDK path")
tracker-hydra/CMakeLists.txt:set(SDK_HYDRA "" CACHE PATH "libSixense path for Razer Hydra")
tracker-kinect-face/CMakeLists.txt:     set(SDK_KINECT20 "$ENV{KINECTSDK20_DIR}" CACHE PATH "Kinect SDK path")
tracker-rs/CMakeLists.txt:set(SDK_REALSENSE "$ENV{RSSDK_DIR}" CACHE PATH "Intel RealSense SDK dir")
tracker-tobii/CMakeLists.txt:    set(SDK_TOBII "" CACHE PATH "Tobii Stream Engine path")
tracker-trackhat/CMakeLists.txt:        set(SDK_TRACKHAT_SENSOR CACHE PATH "")
```
I guess the target is for paeons to undertand how to generate the TrackIR WINE injector so we can slap it in Proton prefixes? 
Or perhaps I need to get this project building with Actions in a way that can be brought into Proton or GE-Proton so it can be automatically present in each prefix?
